### ISSUE TITLE
Easier way to migrate globally installed binaries between node versions

### ISSUE THREAD

**Issue by cafreeman** [volta-cli/volta#935]
Currently, if you've got a globally installed binary (e.g. `ember-cli`), and you switch Node versions, you have to also either re-pin or re-install any binaries might need. Per [the docs](https://docs.volta.sh/advanced/packages), this can be done v0.9.0+ with
```
volta run --node 15 npm i -g ember-cli
```

It would be nice if this could be abstracted a bit into a top-level `migrate` command so that it'd be both more ergonomic and more discoverable from `volta`'s help documentation. Maybe something like `volta migrate <package name> <node version>`?

**Comment by chriskrycho** [comment]
Thank you for filing this! @charlespierce for context, @cafreeman and I were chatting about this and I suggested we write it up, as I imagine it's a thing that'll come up more as folks use it more. It *seems* like there's a fairly straightforward approach here, but I don't doubt if we work through an RFC we'll find a few edge cases. 😏

**Comment by charlespierce** [comment]
Hi @cafreeman, can you expand a little bit on what you mean by the following?
> you have to also either re-pin or re-install any binaries might need
The philosophy behind Volta pinning each global package to a Node version at the time of installation is specifically to prevent needing to re-install everything every time you change default Node versions. The tools should continue to work (using the Node version with which they were installed) even though the default Node has changed.

Also, if you change your default (global) Node version, then you shouldn't need to use `volta run` to get the packages to update. Running `npm i -g ember-cli` will use whatever your _current_ default is, so if you've switched to Node 15 then it will use that and update the pinned version. The `volta run --node <version> npm i -g <package>` workaround is for when you want to pin a package to a version that is _different_ from your current default for whatever reason.

**Comment by chriskrycho** [comment]
Yeah, we talked through this when chatting. The *default* we have today is good, but if you *want* to migrate a bunch of tools to whatever your current pinned version is, that's a bunch more annoying. Related, the idea in #931 could make this more straightforward: you can imagine being able to do `volta list --<some format> | npm i -g` and have it "just work" to switch everything to using a new version of Node for all of them.

One use case: upgrading all the tools in the default toolchain if you have a security vulnerability against a Node version.

**Comment by kevinfiol** [comment]
Glad I found this thread because I was also wondering about this. Volta seems particularly nice for handling global installs. And while this probably wasn't intended to be in Volta's scope, it's kinda replaced nvm for me (even though it's not very convenient as an nvm replacement).

One nice addition to have would be to see which version of Node the global binaries are pinned to. Currently, when I do `volta list`, it just shows all my binaries with `(default)` next to them.

**Comment by zaro** [comment]
While the default makes sense as to keep stuff working, something like what `nvm` has with :

    nvm install 'lts/*' --reinstall-packages-from=current

that allows you install a node version, and also pins all your global packages from the current one will be super helpful when you need  to migrate between node versions.
